mcx/distributions/mvnormal.py

```diff
@@ -0,0 +1,40 @@
+from jax import numpy as np
+from jax import random, scipy
+
+from mcx.distributions import constraints
+from mcx.distributions.distribution import Distribution
+from mcx.distributions.shapes import broadcast_batch_shape
+
+
+class MvNormal(Distribution):
+    params_constraints = {
+        "mu": constraints.real_vector,
+        "covariance_matrix": constraints.positive_definite,
+    }
+    support = constraints.real_vector
+
+    def __init__(self, mu, covariance_matrix):
+        self.event_shape = broadcast_batch_shape(np.shape(mu))
+        self.batch_shape = broadcast_batch_shape(
```

Comment:
And as you implied, we may also want to define iid random variables with the same mean vector but different covariance matrices; `batch_shape` thus also depends on the shape of `covariance_matrix`. But we want to ignore the last dimension of the broadcasted shape since it will correspond to the `event_shape`. So:

`batch_shape = broadcast_batch_shape(np.shape(mu), np.shape(covariance_matrix))[:-2]`

Reply:
Note that we should also check that the shapes of `mu` and `covariance_matrix` are consistent i.e. that `np.shape(covariance_matrix)[-2:-1]` is equal to `event_shape`. And raise an exception if that is not the case.